src/orchestrator.ts

```diff
@@ -25,7 +25,7 @@ export class Orchestrator {
     private newGuidCounter: number;
     private subOrchestratorCounter: number;
 
-    constructor(public fn: (context: IOrchestrationFunctionContext) => IterableIterator<unknown>) { }
+    constructor(public fn: (context: IOrchestrationFunctionContext) => Generator<unknown, unknown, any>) { }
```

Comment:
Is there a reason we went with `any` for `TNext`? My understanding was that we should always use `unknown` instead of `any` whenever possible.

Reply:
Yeah, and it's quite interesting actually. 

So if we were to use `unknown` for `TNext`, TypeScript would force end-users would to use type-assertions (`x as <type>`) after every `yield` statement in their functions / orchestrators / etc. 

This can be evidenced when writing tests, which simulate user code. In the tests, if we had an `unknown` instead of `any`, each `yield` statement would need to followed by a type-assertions (`output as <-type>`) in order to do pretty much anything with the result. Say, for instance, before accessing property from it. I didn't want to do this because it would break people's expected usage of the service, assuming they were using TypeScript too.

This is why I said that the older version of typescript also leaves `TNext` as any / untyped.

Reply:
Makes sense!

Reply:
In the near future, I would love to discuss this with the TypeScript folks because, as you point out, `any` types are an escape hatch from the type-checker.

I imagine some future where you could type your azure functions such that TypeScript would know that each `yield` corresponds to a user-defined type. In that scenario, we could enforce something stricter than `unknown` and `any` without forcing type assertions. It would be so cool!

But that's for another PR, much later in the future 😄